decoder.go

```diff
@@ -66,11 +66,44 @@ func NewDecoder(p []byte) *Decoder {
 	}
 }
 
+// Mode returns the current decoding mode, safe vs fastest.
```

Comment:
can you update the PR title/body to mention this change as well.

Reply:
It's mentioned in the PR description and the body of the commit message

Reply:
My ability to read appears to have regressed in the new year